doc/user/content/transform-data/patterns/rules-engine.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Rules execution engine"
+description: "Encode rules as data and apply them using lateral joins."
+aliases:
+  - /guides/rules-engine/
+  - /sql/patterns/rules-engine/
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+SQL queries are often used to represent "rules" and evaluate which data in a dataset adhere to those rules.
+In Materialize, the results of these queries can be kept automatically up-to-date, enabling use cases like real-time, rules-based alerting.
+However, some organizations have a lot of rules.
+This can lead users to ask questions like "Can Materialize support 100,000 materialized views"?
+
+Fortunately there is a pattern to address this need without creating zillions of separate views.
+If your rules are simple enough to be expressed as data (i.e. not arbitrary SQL), then you can use lateral joins to implement a rules execution engine.
+
+## Hands-on Example
+
+Suppose we have a dataset about birds, and we'd like to subscribe to all the birds who satisfy a set of rules.
+Instead of creating a separate view for each rule, we will encode the rules **as data** and use a `LATERAL` join to execute them.
+
+A `LATERAL` join is essentially a `for` loop; for each element of one dataset, do something with another dataset.
+In our example, for each rule in a `bird_rules` dataset, we filter the `birds` dataset according to the rule.
```

Comment:
```suggestion
In this example, we have a dataset about birds. We need to subscribe to all birds in the dataset that satisfy a set of rules. Instead of creating separate views for each rule, you can encode the rules **as data** and use a `LATERAL` join to execute them.
```

Reply:
I like this as a revision of the first paragraph, but I'd like to keep the second paragraph. It spells out how lateral joins work and how they are used to achieve the goal, which I like.